packages/core/src/api/exceptions/initialize.ts

```diff
@@ -32,7 +38,21 @@ export function initializeExceptions(transports: Transports, metas: Metas, trace
     }
   };
 
+  const pushError: ExceptionsAPI['pushError'] = (error) => {
+    const message = error.message;
+    const opts: PushExceptionOptions = {};
+    if (error.name) {
+      opts.type = error.name;
+    }
+    if (error.stack && config.parseStacktrace) {
+      opts.stackFrames = config.parseStacktrace(error).frames;
+    }
+
+    return pushException(message, opts);
+  };
+
   return {
     pushException,
+    pushError,
```

Comment:
imho, the naming is a little bit confusing. An exception is an error technically and we also have a `pushLog` which can be an error.

Reply:
Error objects are pretty hard to work with as they are non-standard across apps, browsers and platforms.

We can provide a helper method to the users that allow them to manually parse an error and then leverage on `pushExceptions` as more or less this is what we are doing under the hood.

Reply:
I feel strongly about this: 
It's idiomatic to throw `Error` objects in js, and the most common use case will be to report such error afte receiving it through `catch(e) {...}`. Thus we need an easy way to do it. All comparable libs have it as well.

Conversly, the current `pushException` feels like an internal thing. If the user has a non-Error object, probaly a string message, most likely he should use the `api.pushLog` with `error` level. There's not really a situation where you have stack trace, type and message but not an `Error`, other than the `onerror` event, which we already cover.

So to make things less confusing, maybe let's make `pushException` internal?

Reply:
So basically:
- allow other packages and/or the users to pass a stacktrace parser and make `pushException` receive an error object instead of string. The stacktrace would be extracted using the mentioned parser and the type would be extracted from the `name` property on the object.
- for string-based errors just use the logging API from the lib

In this case I don’t see the need for another method at this level tho’. Just enhance the existing one? Maybe add a config property that forces a specific type and stacktrace if somebody actually really wants to override them.

Reply:
It seems like we're on the same page more or less then? Leave one method which consumes `Error`, parser ir configurable, provided automatically if you init via `@grafana/agent-web`, or `@grafana/agent-node` in the future

Reply:
updated